h5netcdf/dimensions.py

```diff
@@ -86,9 +86,15 @@ def __init__(self, parent, name, size=None, create_h5ds=False, phony=False):
             self._root._phony_dim_count += 1
         else:
             self._root._max_dim_id += 1
+
         self._dimensionid = self._root._max_dim_id
-        if parent._root._writable and create_h5ds and not self._phony:
+        if self._phony:
+            self._h5ds = None
+        elif parent._root._writable and create_h5ds:
+            # Create scale will create the self._h5ds object
             self._create_scale()
+        else:
+            self._h5ds = self._root._h5file[self._h5path]
```

Comment:
I think this is actually buggy (and the associated strong reference created in `self._create_scale`). I think you were write careful to make all the strong references short lived. but this is a long lived strong reference to the dataset.

Essentially, this contributes to about 50 ms of time savings because the `self._h5ds` is a pretty slow lookup and is used repeatidely to get the different properties of the dataset.

We could frontload all the properties (dtype, shape, and others) early on, to avoid the cost of lazy creation.

Reply:
The Dimension-class is a relatively new member of h5netcdf. It might need optimization too.

But as always, we need to be careful to mimic the netcdf-c behaviour. Therefore the Dimension-class always asks from underlying h5py/hdf5. If you think you've found a way how to avoid that, very good!